File: src/main/java/com/codeperfection/shipit/dto/product/ProductDto.java
package com.codeperfection.shipit.dto.product;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProductDto {

    private UUID uuid;

    private String name;

    private Integer volume;

    private Integer price;

    private Integer countInStock;
}

File: src/main/kotlin/com/codeperfection/shipit/dto/product/ProductDto.kt
package com.codeperfection.shipit.dto.product

import com.codeperfection.shipit.entity.Product
import java.util.*

data class ProductDto(
    val id: UUID,
    val userId: UUID,
    val name: String,
    val volume: Int,
    val price: Int,
    val countInStock: Int
) {
    companion object {
        fun fromEntity(product: Product) = ProductDto(
            id = product.id,
            userId = product.userId,
            name = product.name,
            volume = product.volume,
            price = product.price,
            countInStock = product.countInStock
        )
    }
}
